styx-scheduler-service/src/test/java/com/spotify/styx/state/StateUtilTest.java

```diff
@@ -113,4 +115,22 @@ public void shouldGetUsedResourcesStorageException() throws IOException {
     StateUtil.getResourcesUsageMap(storage, timeoutConfig, workflowCache,
         Instant.ofEpochMilli(11L), resourceDecorator);
   }
+
+  @Test
+  public void shouldConsumeResource() {
+    assertTrue(StateUtil.isConsumingResources(RunState.State.PREPARE));
+    assertTrue(StateUtil.isConsumingResources(RunState.State.SUBMITTING));
+    assertTrue(StateUtil.isConsumingResources(RunState.State.SUBMITTED));
+    assertTrue(StateUtil.isConsumingResources(RunState.State.RUNNING));
+    assertTrue(StateUtil.isConsumingResources(RunState.State.TERMINATED));
+    assertTrue(StateUtil.isConsumingResources(RunState.State.FAILED));
+    assertTrue(StateUtil.isConsumingResources(RunState.State.ERROR));
+    assertTrue(StateUtil.isConsumingResources(RunState.State.DONE));
```

Comment:
Wait, I think exclude ERROR and DONE... If we never transition out of them, the resources will never decrement... In the old code we decremented resources implicitly by deleting the ActiveWorkflowInstance.

We should not literally be "counting terminal states" but "counting states where the container is not yet cleaned up"...